source/darwin/secure_transport_tls_channel_handler.c

```diff
@@ -567,6 +588,51 @@ static int s_handle_shutdown(
             "id=%p: shutting down read direction with error %d. Flushing queues.",
             (void *)handler,
             error_code);
+        if (!abort_immediately && secure_transport_handler->negotiation_finished &&
+            !aws_linked_list_empty(&secure_transport_handler->input_queue) && slot->adj_right) {
```

Comment:
don't think you need to check `adj_right`?

`s_process_read_message()` deals with it just fine. If it's NULL, it uses SIZE_MAX as the window, and discards the aws_io_message after invoking the TLS on_data_read() callback